qiskit_nature/drivers/second_quantization/fermionic_molecule_driver.py

```diff
@@ -0,0 +1,169 @@
+            PySCFDriver.check_installed()
+            driver_class = PySCFDriver
+        elif driver_type == FermionicDriverType.PSI4:
+            PSI4Driver.check_installed()
+            driver_class = PSI4Driver
+        elif driver_type == FermionicDriverType.PYQUANTE:
+            PyQuanteDriver.check_installed()
+            driver_class = PyQuanteDriver
+        elif driver_type == FermionicDriverType.GAUSSIAN:
+            GaussianDriver.check_installed()
+            driver_class = GaussianDriver
+        else:
+            MissingOptionalLibraryError(libname=driver_type, name="FermionicDriverType")
+
+        logger.debug("%s found from type %s.", driver_class.__name__, driver_type.value)
+        return driver_class
+
+
+class FermionicMoleculeDriver(FermionicDriver):
+    """
+    Molecule based fermionic driver
+    """
+
+    def __init__(
+        self,
+        molecule: Molecule,
+        basis: str = "sto3g",
+        method: MethodType = MethodType.RHF,
+        driver_type: FermionicDriverType = FermionicDriverType.AUTO,
+        driver_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        Args:
+            molecule: molecule
+            basis: basis set
+            method: The method type to be used for the calculation.
+            driver_type:type of driver to be used. If `AUTO` is selected, it will use
+                        the first driver installed in the following order:
+                        `PYSCF`, `PSI4`, `PYQUANTE`, `GAUSSIAN`
+            driver_kwargs: kwargs to be passed to driver
+
+        Raises:
+            MissingOptionalLibraryError: Driver not installed.
+        """
+        super().__init__()
+        self._driver_class = FermionicDriverType.driver_class_from_type(driver_type)
+        self._driver_kwargs = driver_kwargs
+        self._molecule = molecule
+        self._basis = basis
+        self._method = method
+
+    @property
+    def molecule(self) -> Optional[Molecule]:
+        """return molecule"""
+        return self._molecule
+
+    @molecule.setter
+    def molecule(self, value: Molecule) -> None:
+        """set molecule"""
+        self._molecule = value
+
+    @property
+    def basis(self) -> str:
+        """return basis"""
+        return self._basis
+
+    @basis.setter
+    def basis(self, value: str) -> None:
+        """set basis"""
+        self._basis = value
+
+    @property
+    def method(self) -> MethodType:
+        """return Hartree-Fock method"""
+        return self._method
+
+    @method.setter
+    def method(self, value: MethodType) -> None:
+        """set Hartree-Fock method"""
+        self._method = value
+
+    @property
+    def driver_kwargs(self) -> Optional[Dict[str, Any]]:
+        """return driver kwargs"""
+        return self._driver_kwargs
+
+    @driver_kwargs.setter
+    def driver_kwargs(self, value: Optional[Dict[str, Any]]) -> None:
+        """set driver kwargs"""
+        self._driver_kwargs = value
```

Comment:
I am wondering whether it would make sense to define a small `MoleculeDriver` interface which would take care of all these properties (which are identical in here and in `BosonicMoleculeDriver`). This would also allow type checking against the interface in case we ever need to differentiate a `MoleculeDriver` from another one (say e.g. in the `BopesSampler`)

Reply:
I had discussed that with Steve but it would need to be a separate interface not derived from BaseDriver. so it would be 2 base classes for those and you wouldn't be able to use as a typehint anyway because it is not a BaseDriver

Reply:
What is the problem with using two base classes?

Reply:
The idea was also whether we could extract the duplication of the attributes like `basis`, `method`, etc.
But I guess that is fine as is..